wasi-ld now links libxml2 but it is not in wasi-sdk

Testing on a container with minimal packages.  I installed wasi-sdk 16.0, set it in my $PATH, and compiled a simple C program to wasm (https://github.com/singlestore-labs/singlestore-wasm-toolkit/tree/main/examples/cpp/power).  Compilation failed with:

```
clang++                             \
    -fno-exceptions                 \
    --target=wasm32-unknown-wasi    \
    -mexec-model=reactor            \
    -s                              \
    -I.                             \
    -o split.wasm                   \
    split.cpp
/opt/wasi-sdk/bin/wasm-ld: error while loading shared libraries: libxml2.so.2: cannot open shared object file: No such file or directory
clang-14: error: unable to execute command: No such file or directory
clang-14: error: linker command failed due to signal (use -v to see invocation)
make: *** [Makefile:4: split.wasm] Error 255
```

The test program doesn't use libxml2 at all, so I'm guessing wasi-ld is implicitly trying to include it.  Indeed, when I installed the libxml2-dev package into my container, compilation was successful.

One more data point -- this compilation succeeded in wasi sdk 14.0.

Does libxml2 now need to be included in wasi sdk?

--- a/Makefile
+++ b/Makefile
@@ -66,6 +66,7 @@ build/llvm.BUILT:
 			  -DDEFAULT_SYSROOT=$(PREFIX)/share/wasi-sysroot), \
 		     -DDEFAULT_SYSROOT=$(PREFIX)/share/wasi-sysroot) \
 		-DLLVM_INSTALL_BINUTILS_SYMLINKS=TRUE \
+		-DLLVM_ENABLE_LIBXML2=OFF \
 		$(LLVM_PROJ_DIR)/llvm
 	DESTDIR=$(DESTDIR) ninja $(NINJA_FLAGS) -C build/llvm \
 		install-clang \

